feat: add `lapack/base/dlacn2`

aayush0325 opened the pull request:
## Description

> What is the purpose of this pull request?

This pull request:

-   adds `lapack/base/dlacn2`

## Related Issues

> Does this pull request have any related issues?

No.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

aayush0325 committed:
feat: add lapack/base/dlacn2

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
feat: add main exports

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
feat: add examples and benchmarks

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
test: initial tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
test: add more tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
test: add test for N = 1

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
test: more tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
test: update tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
test: add offset tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
test: write tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
docs: add ts files

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
docs: update jsdoc

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
docs: add readme

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
docs: add repl.txt

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/README.md:
```diff
@@ -0,0 +1,362 @@
+
+#### dlacn2( N, V, X, ISGN, EST, KASE, ISAVE )
+
+Estimates the one-norm of a square matrix `A`, using alternative indexing semantics and reverse communication for evaluating matrix-vector products.
```
```suggestion
Estimates the one-norm of a square matrix `A`, using reverse communication for evaluating matrix-vector products.
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/README.md:
```diff
@@ -0,0 +1,362 @@
+The function has the following parameters:
+
+-   **N**: number of rows/columns in `A`.
+-   **V**: workspace [`Float64Array`][mdn-float64array] having `N` indexed elements, used internally to store intermediate vectors.
+-   **X**: input/output [`Float64Array`][mdn-float64array] having `N` indexed elements, contains the current or next matrix-vector product.
+-   **ISGN**: [`Int32Array`][mdn-int32array] having `N` indexed elements, stores the sign of each element in `X` during iterations.
+-   **EST**: single-element [`Float64Array`][mdn-float64array], on output, contains the estimated one-norm of the matrix `A`.
+-   **KASE**: single-element [`Int32Array`][mdn-int32array] that controls the reverse communication.
+-   **ISAVE**: [`Int32Array`][mdn-int32array] having 3 indexed elements, used internally to maintain state across multiple calls.
```
```suggestion
-   **V**: workspace [`Float64Array`][mdn-float64array] having `N` indexed elements. This workspace is used internally to store intermediate vectors.
-   **X**: input/output [`Float64Array`][mdn-float64array] having `N` indexed elements. This array contains the current or next matrix-vector product.
-   **ISGN**: [`Int32Array`][mdn-int32array] having `N` indexed elements. This array stores the sign of each element in `X` during iterations.
-   **EST**: single-element [`Float64Array`][mdn-float64array], on output. This array contains the estimated one-norm of the matrix `A`.
-   **KASE**: single-element [`Int32Array`][mdn-int32array] that controls the reverse communication.
-   **ISAVE**: [`Int32Array`][mdn-int32array] having 3 indexed elements. This array is used internally to maintain state across multiple calls.
```

kgryte replied:
Here and below.

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/README.md:
```diff
@@ -0,0 +1,362 @@
+-   **KASE**: single-element [`Int32Array`][mdn-int32array] that controls the reverse communication.
+-   **ISAVE**: [`Int32Array`][mdn-int32array] having 3 indexed elements, used internally to maintain state across multiple calls.
+
+The reverse communication takes place using `KASE`, it may have any of these values:
```
```suggestion
The reverse communication takes place using `KASE`. The single-element in `KASE` may have the following values:
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/lib/dlacn2.js:
```diff
@@ -0,0 +1,80 @@
+*
+* ## Notes
+*
+* the reverse communication takes place using `KASE`, it may have any of these values:
```
```suggestion
* The reverse communication takes place using `KASE`, it may have any of these values:
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/lib/ndarray.js:
```diff
@@ -0,0 +1,90 @@
+*
+* ## Notes
+*
+* the reverse communication takes place using `KASE`, it may have any of these values:
```
```suggestion
* The reverse communication takes place using `KASE`. The single-element in `KASE` may have the following values:
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/lib/nint.js:
```diff
@@ -0,0 +1,65 @@
+* var out = nint( 4.5 );
+* // returns 4.0
+*/
+function nint( X ) {
```
TODO: replace once Karan's PR for `round-nearest-even` gets merged.

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/lib/base.js:
```diff
@@ -0,0 +1,487 @@
+var nint = require( './nint.js' );
+
+
+// MAIN //
```
```suggestion
// FUNCTIONS //
```

And then add `// MAIN //` below for the main export.

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/lib/base.js:
```diff
@@ -0,0 +1,487 @@
+
+	jlast = ISAVE[ offsetISAVE + strideISAVE ];
+	ISAVE[ offsetISAVE + strideISAVE ] = idamax( N, X, strideX, offsetX );
+
```
I suggest making your expressions a bit tighter using temporary variables here for indices.

```javascript
var ii1 = offsetISAVE + strideISAVE;
var ii2 = ii1 + strideISAVE;

// ...

ISAVE[ ii1 ] && ISAVE[ ii2 ]

// ...

vi = ISAVE[ ii1 ];

// ...

X[ offsetX + ( vi * strideX ) ]
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/lib/base.js:
```diff
@@ -0,0 +1,487 @@
+	ISAVE[ offsetISAVE + ( 2 * strideISAVE ) ] = 2;
+
+	ix = offsetX;
+	for ( i = 0; i < N; i++ ) {
```
This could be replaced with `blas/ext/base/dfill`.

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/test/test.dlacn2.js:
```diff
@@ -0,0 +1,222 @@
+		if ( KASE[ 0 ] === 0 ) {
+			break;
+		}
+		else if ( KASE[ 0 ] === 1 ) {
```
```suggestion
		} else if ( KASE[ 0 ] === 1 ) {
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/test/test.dlacn2.js:
```diff
@@ -0,0 +1,222 @@
+		}
+	}
+
+	expectedEST = new Float64Array( [ 25.0 ] );
```
```suggestion
	// Expected values determined via the reference Fortran routines in LAPACK:
	expectedEST = new Float64Array( [ 25.0 ] );
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/test/test.dlacn2.js:
```diff
@@ -0,0 +1,222 @@
+		if ( KASE[ 0 ] === 0 ) {
+			break;
+		}
+		else if ( KASE[ 0 ] === 1 ) {
```
```suggestion
		} else if ( KASE[ 0 ] === 1 ) {
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/test/test.dlacn2.js:
```diff
@@ -0,0 +1,222 @@
+		if ( KASE[ 0 ] === 0 ) {
+			break;
+		}
+		else if ( KASE[ 0 ] === 1 ) {
```
```suggestion
		} else if ( KASE[ 0 ] === 1 ) {
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/test/test.ndarray.js:
```diff
@@ -0,0 +1,588 @@
+			break;
+		}
+		else if ( KASE[ 1 ] === 1 ) {
+			dgemv( 'no-transpose', 4, 4, 1.0, A, 4, 1, 0, X, 1, 1, 0, work, 1, 1 );
```
```suggestion
			// A*x:
			dgemv( 'no-transpose', 4, 4, 1.0, A, 4, 1, 0, X, 1, 1, 0, work, 1, 1 );
```

kgryte replied:
Is `work` strictly necessary here? Can we write directly to `X` with `dgemv`?

kgryte replied:
After looking at `dgemv`, agree with the choice to use a `work` array. Adding a comment still stands.

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/test/test.ndarray.js:
```diff
@@ -0,0 +1,588 @@
+			dgemv( 'no-transpose', 4, 4, 1.0, A, 4, 1, 0, X, 1, 1, 0, work, 1, 1 );
+			dcopy( 4, work, 1, 1, X, 1, 1 );
+		} else if ( KASE[ 1 ] === 2 ) {
+			dgemv( 'transpose', 4, 4, 1.0, A, 4, 1, 0, X, 1, 1, 0, work, 1, 1 );
```
```suggestion
			// A^T * x:
			dgemv( 'transpose', 4, 4, 1.0, A, 4, 1, 0, X, 1, 1, 0, work, 1, 1 );
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlacn2/test/test.dlacn2.js:
```diff
@@ -0,0 +1,222 @@
+
+	work = new Float64Array( 1 );
+
+	while ( true ) {
```
May be good to add a comment before the `while` loop explaining, conceptually, what you are doing here.

aayush0325 commented:
/stdib merge

aayush0325 commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into lapack-dlacn2

aayush0325 committed:
refactor: add comments and optimise pointer arithmetic

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
chore: cleaning up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
chore: cleanup

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
chore: cleanup

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 commented on lib/node_modules/@stdlib/lapack/base/dlacn2/test/test.dlacn2.js:
```diff
@@ -0,0 +1,262 @@
+			dcopy( 4, work, 1, X, 1 );
+		} else if ( KASE[ 0 ] === 2 ) {
+			/*
+			* -   If `KASE` == 2: `dlacn2` wants us to compute A ^ T * X and overwrite X.
```
```suggestion
			* -   If `KASE` == 2: `dlacn2` wants us to compute A ^ T * X and overwrite X.
```
the `*` shouldn't be here

aayush0325 commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into lapack-dlacn2

aayush0325 committed:
chore: cleanup

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown_pkg_readmes
    status: passed
  - task: lint_markdown_docs
    status: na
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---
